### ISSUE TITLE
support "proxy" endpoint  when dragging

### ISSUE THREAD

**Issue by sporritt** [sporritt/jsPlumb#22]
from google code

https://code.google.com/p/jsplumb/issues/detail?id=236

**Comment by sporritt** [comment]
...before google code shuts down, here's what this issue is about

from this discussion on the groups:

https://groups.google.com/forum/?fromgroups#!topic/jsplumb/HtdSANpiDuk

It would be good to support the notion of a "proxy" endpoint, for jsPlumb to use when dragging a connection (new or existing).  At the moment, when you drag a connection from some endpoint, jsPlumb uses a clone of that endpoint.  If the endpoint is large this can make for an unappealing user experience.

of course there are other reasons why you might want to use this notion of a drag proxy.  i mention that one because it was the catalyst for opening this issue.

possible syntax :

jsPlumb.addEndpoint("el", {
  isSource:true,
  endpoint:["Dot",{radius:40}],
  paintStyle:{fillStyle:"blue"},
  proxy:{
    endpoint:["Dot",{radius:5}],
    paintStyle:{fillStyle:"red"}
  }
});

It would seem also to be a good idea to add the notion of a 'DragProxy' to the jsPlumb defaults while we're at it.

**Comment by sporritt** [comment]
fixed in 2.0.0 dev. you can now pass a `dragProxy` endpoint definition to either `addEndpoint` or `makeSource`:

```
jsPlumb.addEndpoint("el", {
  isSource:true,
  endpoint:["Dot",{radius:40}],
  paintStyle:{fillStyle:"blue"},
  dragProxy:[ "Dot", { radius:5, cssClass:"styleMe" } ]
});
```

note the extended syntax with paint style was dropped as 2.0.0 doesnt support vml and so if needs be a `cssClass` can be used to style the endpoint.
